src/app/modules/app/app.js

```diff
@@ -123,18 +139,106 @@
-//            $scope.topLanguages = [
-//                {isoCode: 'en', name:'English'},
-//                {isoCode: 'tpi', name:'Tok Pisin'},
-//                {isoCode: 'th', name:'Thai'},
-//                {isoCode: 'id', name:'Bahasia Indonesia'},
-//                {isoCode: 'fr', name:'French'}
-//            ];
             languageService.getLanguages().then(function(languages) {
-                $scope.topLanguages = languages;
+                //This is the number of displayed languages on the browse sidebar before (n more...)
+                var numberOfTopLanguages = 4;
+
+                function compareLang(a, b) {
+                    return a.name < b.name ? -1 : 1;
+                }
+
+                //If all languages can be held in the top list, just fill the top list
+                if(languages.length <= numberOfTopLanguages) {
+                    $scope.topLanguages = languages.sort(compareLang);
+                    $scope.otherLanguages = [];
+                }
+                else {
+                    //Split the list of languages into top and other, and sort the lists alphabetically
+                    $scope.topLanguages = languages.slice(0, numberOfTopLanguages).sort(compareLang);
+                    $scope.otherLanguages = languages.slice(numberOfTopLanguages, languages.length).sort(compareLang);
+                }
+
+                for(var i = 0; i < $scope.otherLanguages.length; i++) {
+                    if($scope.otherLanguages[i].isoCode == $scope.currentLang) {
+                        $scope.topLanguages.unshift($scope.otherLanguages.splice(i, 1)[0]);
+                        $scope.topLanguages.sort(compareLang);
+                        i = $scope.otherLanguages.length;
+                    }
+                }
             });
-            $scope.topTags = tagService.getTags();
+
+                //This is the global list of all categories of tags
+                //The tags are separated out by these categories on the sidebar of the browse view
+                //id is the usage in the tag (e.g. "region.MyRegion") and displayName is the header
+                $scope.tagCategories = [
+                    {id: 'topic', displayName: 'Topics'},
+                    {id: 'region', displayName: 'Regions'},
+                    {id: 'level', displayName: 'Reading Levels'}
+                ];
+
+                //This is the object which will hold all of the tag names
+                $scope.tags = {topic: {top: [], other: []}};
+
+                tagService.getTags().then(function(tags) {
+                    //Get the names out of the tags; we don't care about the other properties
+                    var tagNames = tags.map(function(item) {
+                        return item.name;
+                    });
+
+                    //This is the number of tags that will be shown in each category of tag before the (n more...)
+                    var numberOfTopTags = 100;
```

Comment:
OK, we should have a comment to that effect. Also, I suspect the HTML that would actually show the n-more hasn't been done? The comment should mention that and give an idea where the change would be needed. I'd also be inclined to kick the number up to 1000; it's not inconceivable we could get 100 if we start letting users add them.